components/contactUs.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState, useRef, useEffect } from "react";
+import { RenderMsg } from "./renderMsg";
+
+type FormState = {
+  name: string;
+  email: string;
+  message: string;
+};
+
+export const ContactFrom = () => {
+  const [nameErr, setNameErr] = useState("");
+  const [emailErr, setEmailErr] = useState("");
+  const [messageErr, setMessageErr] = useState("");
+  const [state, setstate] = useState<FormState>({
+    name: "",
+    email: "",
+    message: "",
+  });
+
+  const nameInput = useRef<HTMLInputElement>(null);
+  const emailInput = useRef<HTMLInputElement>(null);
+  const messageInput = useRef<HTMLTextAreaElement>(null);
+
+  useEffect(() => {
+    if (nameInput.current != null) {
+      nameInput.current.focus();
+    }
+  }, []);
+
+  const setFocus = (
+    ref:
+      | React.RefObject<HTMLInputElement>
+      | React.RefObject<HTMLTextAreaElement>
+  ) => {
+    if (ref.current != null) {
+      ref.current.focus();
+    }
+  };
+
+  const handleChange = (
+    event:
+      | React.ChangeEvent<HTMLInputElement>
+      | React.ChangeEvent<HTMLTextAreaElement>
+  ) => {
+    setstate({ ...state, [event.target.name]: event.target.value });
+  };
+
+  const formIsValid = () => {
+    let validRegex =
```

Comment:
Make it a habit to declare variables as `const` by default and convert to `let` if necessary. This can help prevent weird bugs

Reply:
What do you mean by "converting" to let?